### PR TITLE
Fix/2120

### LINKED ISSUE #2120: storage.Delete not supported on Android/iOS
### Describe the bug:
This bug is related to bug "File-I/O on Android #2118"

In order to create a work around for bug 2118, I tried to delete the existing file before writing the new data to it.
The method "storage.Delete (uri)" produces on Android the error
"operation not supported for this URI"

### To Reproduce:
Steps to reproduce the behaviour:

### Device (please complete the following information):
 - **OS:**   Android
 - **Version:**   Android 10
 - **Go version:**   1.15.6
 - **Fyne version:**   Develop branch, pre 2.0.2

### PR DESCRIPTION
**Pull request by andydotxyz**
Fixes #2120

### Checklist:

- [ ] Tests included. <- tested with the same production code as the iOS work
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: internal/driver/mobile/android.c [hunk 1]
```diff
@@ -328,13 +328,37 @@ const char* contentURIGetFileName(uintptr_t jni_env, uintptr_t ctx, char* uriCst
...
 	size_t length = strlen(uriCstr)-7;// -7 for 'file://'
 	char* path = malloc(sizeof(char)*(length+1));// +1 for '\0'
 	memcpy(path, &uriCstr[7], length);
 	path[length] = '\0';
 
+    return path;
+}
+
+bool deleteFileURI(char *uriCstr) {
+    char* path = filePath(uriCstr);
```
Comment (Jacalz) [inline internal/driver/mobile/android.c:342]: Indentation looks strange here

File: internal/driver/mobile/android.c [hunk 2]
```diff
@@ -328,13 +328,37 @@ const char* contentURIGetFileName(uintptr_t jni_env, uintptr_t ctx, char* uriCst
...
+	if (!hasPrefix(uriCstr, "file://")) {
+	    LOG_FATAL("Cannot delete for scheme: %s", uriCstr);
+		return false;
+	}
+
+	return deleteFileURI(uriCstr);
+}
+
+bool existsFileURI(char* uriCstr) {
+    char* path = filePath(uriCstr);
```
Comment (Jacalz) [inline internal/driver/mobile/android.c:360]: Same as above.

**Review by Jacalz — CHANGES_REQUESTED** [review]
Looks good. Just some notes on what looks like incorrect indentation

File: internal/driver/mobile/android.c (hunk 1 above)
Comment (andydotxyz) [reply internal/driver/mobile/android.c]: Thanks @Jacalz - seems that GoLand's C indentation isn't totally consistent...

**Review by Jacalz — APPROVED** [review]

**Pull request merged**